pkgs/by-name/te/tetrio-desktop/package.nix

```diff
@@ -40,45 +24,39 @@ stdenv.mkDerivation (finalAttrs: {
 
   nativeBuildInputs = [
     dpkg
-    autoPatchelfHook
-    wrapGAppsHook
-  ];
-
-  dontWrapGApps = true;
-
-  buildInputs = [
-    alsa-lib
-    cups
-    libX11
-    libXScrnSaver
-    libXtst
-    mesa
-    nss
-    gtk3
+    makeWrapper
   ];
 
-  unpackCmd = "dpkg -x $curSrc src";
-
   installPhase = ''
     runHook preInstall
 
-    mkdir -p $out/bin
-    cp -r opt/ usr/share/ $out
-    ln -s $out/opt/TETR.IO/TETR.IO $out/bin/tetrio
+    mkdir -p $out
+    cp -r usr/share/ $out
+
+    mkdir -p $out/share/TETR.IO/
+    cp opt/TETR.IO/resources/app.asar $out/share/TETR.IO/app.asar
 
     substituteInPlace $out/share/applications/TETR.IO.desktop \
       --replace-fail "Exec=/opt/TETR.IO/TETR.IO" "Exec=$out/bin/tetrio"
 
     runHook postInstall
   '';
 
-  postFixup = ''
-    wrapProgram $out/opt/TETR.IO/TETR.IO \
-      --prefix LD_LIBRARY_PATH : ${libPath}:$out/opt/TETR.IO \
-      ''${gappsWrapperArgs[@]}
-  '';
+  postFixup =
+    let
+      libPath = lib.makeLibraryPath [
+        libpulseaudio
+      ];
+    in
+    ''
+      makeShellWrapper '${lib.getExe electron}' $out/bin/tetrio \
+        --prefix LD_LIBRARY_PATH : ${addOpenGLRunpath.driverLink}/lib:${libPath} \
+        --add-flags "\''${NIXOS_OZONE_WL:+\''${WAYLAND_DISPLAY:+--ozone-platform-hint=auto --enable-features=WaylandWindowDecorations}}" \
+        --add-flags $out/share/TETR.IO/app.asar
+    '';
```

Comment:
Audio works without `libpulseaudio`, so it can be removed from here and the inputs :)